src/consensus.rs

```diff
@@ -320,24 +320,21 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
                             messages::append_entries_response_inconsistent_prev_entry(self.current_term(),
                                 leader_prev_log_index)
                         } else {
-                            let entries = request.get_entries().unwrap();
-                            let num_entries = entries.len();
-                            scoped_debug!("AppendEntriesRequest: {} entries from leader: {}",
-                                          num_entries, from);
-                            if num_entries > 0 {
-                                let mut entries_vec = Vec::with_capacity(num_entries as usize);
-                                for i in 0..num_entries {
-                                    let entry = entries.get(i);
-                                    let term = Term::from(entry.get_term());
-                                    let data = entry.get_data().unwrap();
-                                    entries_vec.push((term, data));
-                                }
+                            if let Ok(entries) = request.get_entries() {
```

Comment:
check out the `else` branch below (added after your comment). Does that work for you? `unwrap()` panics are notoriously difficult to debug (for example, `rustfmt src/lib.rs` currently crashes for `raft-rs`, but I have no idea where to even start looking because it doesn't unwind the stack to where the panic was triggered).